### PR TITLE
Gcw 1955 add request to order

### PR DESCRIPTION
**Pull request by Darshan-Chauhan**
Hi Team,

This PR adds request feature for Orders. 
Jira ticket:- https://jira.crossroads.org.hk/browse/GCW-1955

please take a look and let me know if further changes are required.
Thanks

### CODE REVIEW

File: app/controllers/api/v1/goodcity_requests_controller.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+      api :DELETE, "/v1/goodcity_requests/1", "Delete goodcity_request"
+      def destroy
+        @goodcity_request.destroy
+        render json: {}
+      end
+
+      private
+
+      def serializer
```
Comment (abulsayyad123) [inline app/controllers/api/v1/goodcity_requests_controller.rb]: Do we need 'serializer' method here?. Serializer creation is managed in ApiController by 'save_and_render_object_with_errors' method .

Comment (Darshan-Chauhan) [reply app/controllers/api/v1/goodcity_requests_controller.rb]: No we don't @abulsayyad123 . I just kept it there for future use.

File: app/models/ability.rb [hunk 2]
```diff
@@ -6,7 +6,7 @@ class Ability
 
   attr_accessor :user, :user_id, :admin, :supervisor, :reviewer, :user_offer_ids, :user_permissions
 
-  PERMISSION_NAMES = ['can_manage_items', 'can_manage_packages',
+  PERMISSION_NAMES = ['can_manage_items', 'can_manage_requests', 'can_manage_packages',
```
Comment (namrataukirde) [inline app/models/ability.rb]: Can we rename it to `can_mange_goodcity_requests` from `can_manage_requests` ?

File: app/models/goodcity_request.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+class GoodcityRequest < ActiveRecord::Base
+  has_paper_trail class_name: 'Version'
+  belongs_to :package_type
+  belongs_to :order
+  belongs_to :created_by, class_name: 'User'
+  validates  :quantity,  numericality: { greater_than_or_equal_to: 1 }
```
Comment (namrataukirde) [inline app/models/goodcity_request.rb]: We can resolve this issue: Unnecessary spacing detected.

File: app/serializers/api/v1/goodcity_request_serializer.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
+module Api::V1
+  class GoodcityRequestSerializer < ApplicationSerializer
+    embed :ids, include: true
+    attributes :id, :quantity, :description, :code_id
+    # Commented below line as it goes into infinite loop of rendering order and request, also don't need to render order again
```
Comment (namrataukirde) [inline app/serializers/api/v1/goodcity_request_serializer.rb]: So if its not needed then can we remove it ?

File: spec/controllers/api/v1/goodcity_requests_contoller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+      expect(response.status).to eq(201)
+    end
+  end
+
+  describe "PUT goodcity_request/1 : update goodcity_request" do
+    before { generate_and_set_token(user) }
+    let(:gc_request) { create(:goodcity_request, quantity: 5) }
+
+    it "Updates goodcity_request record", :show_in_doc do
+      params = goodcity_request_params
```
Comment (namrataukirde) [inline spec/controllers/api/v1/goodcity_requests_contoller_spec.rb]: I think we can remove it. looks like its not used. Correct me if I am wrong. :)

File: app/models/order.rb [hunk 6]
```diff
@@ -18,6 +18,7 @@ class Order < ActiveRecord::Base
   belongs_to :stockit_local_order, -> { joins("inner join orders on orders.detail_id = stockit_local_orders.id and (orders.detail_type = 'LocalOrder' or orders.detail_type = 'StockitLocalOrder')") }, foreign_key: 'detail_id'
 
   has_many :packages
+  has_many :goodcity_requests
```
Comment (namrataukirde) [inline app/models/order.rb]: We can add spec for this association.

File: app/models/package_type.rb [hunk 7]
```diff
@@ -18,6 +18,7 @@ class PackageType < ActiveRecord::Base
 
   has_many :items, inverse_of: :package_type
   has_many :packages, inverse_of: :package_type
+  has_many :goodcity_requests
```
Comment (namrataukirde) [inline app/models/package_type.rb]: We can add spec for this association.

**Comment by Darshan-Chauhan** [comment]
@namrataukirde , added the changes requested.

**Comment by namrataukirde** [comment]
👍

**Pull request merged**